backend/src/libs/packages/server-application/libs/helpers/create-error-response/create-error-response.helper.ts

```diff
@@ -0,0 +1,60 @@
+import { type FastifyError } from 'fastify';
+
+import { ServerErrorType } from '#libs/enums/enums.js';
+import {
+  AuthError,
+  type ValidationError,
+} from '#libs/exceptions/exceptions.js';
+import { HTTPCode, HTTPError } from '#libs/packages/http/http.js';
+import { type ServerErrorResponse, type ValueOf } from '#libs/types/types.js';
+
+type ErrorParameter = FastifyError | ValidationError | HTTPError;
+
+type CreateErrorResponseReturn = {
+  status: ValueOf<typeof HTTPCode>;
+  response: ServerErrorResponse;
+};
+
+const createErrorResponse = (
+  error: ErrorParameter,
+): CreateErrorResponseReturn => {
+  const { message } = error;
+  let status: ValueOf<typeof HTTPCode> = HTTPCode.INTERNAL_SERVER_ERROR;
+  let errorType: ValueOf<typeof ServerErrorType> = ServerErrorType.COMMON;
+  let response: ServerErrorResponse;
+
+  if (error instanceof AuthError) {
+    status = HTTPCode.UNAUTHORIZED;
+    errorType = ServerErrorType.AUTHORIZATION;
```

Comment:
let's make a return from each if case

Reply:
If you know maube better solution of current abstraction without if and switch you are welcome(

Reply:
What comes to my mind first, is that we need to add status and error type in the error instance itself. But as I understood, it is not quite a good option

Reply:
I found one solution. I'll try to do it and we'll see